### PR TITLE
Set context actual domain id

### PR DESCRIPTION
**Pull request by ivanpauno**
See https://github.com/ros2/rcl/pull/718

### CODE REVIEW

File: rmw_fastrtps_shared_cpp/src/participant.cpp [hunk 1]
```diff
@@ -166,11 +166,9 @@ rmw_fastrtps_shared_cpp::create_participant(
 
   // No custom handling of RMW_DEFAULT_DOMAIN_ID. Simply use a reasonable domain id.
 #if FASTRTPS_VERSION_MAJOR < 2
-  participantAttrs.rtps.builtin.domainId =
-    static_cast<uint32_t>(domain_id != RMW_DEFAULT_DOMAIN_ID ? domain_id : 0u);
+  participantAttrs.rtps.builtin.domainId = domain_id;
```
Comment (hidmic) [inline rmw_fastrtps_shared_cpp/src/participant.cpp:169]: @ivanpauno here and below, consider keeping those `static_cast`s.

Comment (ivanpauno) [reply rmw_fastrtps_shared_cpp/src/participant.cpp]: Good point, I didn't see them.

File: rmw_fastrtps_cpp/src/rmw_init.cpp [hunk 2]
```diff
@@ -93,13 +93,13 @@ rmw_init(const rmw_init_options_t * options, rmw_context_t * context)
     return RMW_RET_INVALID_ARGUMENT;
   }
 
-  const rmw_context_t zero_context = rmw_get_zero_initialized_context();
-  assert(0 == std::memcmp(context, &zero_context, sizeof(rmw_context_t)));
```
Comment (hidmic) [inline rmw_fastrtps_cpp/src/rmw_init.cpp:97]: @ivanpauno also unrelated, and can be backtracked to the discussion that took place about UB in `rmw_init()`.

File: rmw_fastrtps_dynamic_cpp/src/rmw_init.cpp [hunk 3]
```diff
@@ -93,13 +93,13 @@ rmw_init(const rmw_init_options_t * options, rmw_context_t * context)
     return RMW_RET_INVALID_ARGUMENT;
   }
 
-  const rmw_context_t zero_context = rmw_get_zero_initialized_context();
-  assert(0 == std::memcmp(context, &zero_context, sizeof(rmw_context_t)));
```
Comment (hidmic) [inline rmw_fastrtps_dynamic_cpp/src/rmw_init.cpp:97]: @ivanpauno same as above.

File: rmw_fastrtps_cpp/src/rmw_init.cpp (hunk 2 above)
Comment (ivanpauno) [reply rmw_fastrtps_cpp/src/rmw_init.cpp:97]: See https://github.com/ros2/rmw_connext/pull/443#discussion_r458314262

**Pull request merged**